feature-riko-BJS2-65650

DmitryRogulkin opened the pull request.

DmitryRogulkin committed:
https://faang-school.atlassian.net/browse/BJS2-65650

config redis + add EventStartEventPublisher

DmitryRogulkin committed:
https://faang-school.atlassian.net/browse/BJS2-65650

config redis + add EventStartEventPublisher

DmitryRogulkin committed:
https://faang-school.atlassian.net/browse/BJS2-65650

Update + code style

DmitryRogulkin committed:
https://faang-school.atlassian.net/browse/BJS2-65650

Final correct.

antikler commented on src/main/java/school/faang/user_service/scheduler/EventStartScheduler.java:
```diff
@@ -0,0 +1,50 @@
+    private final UserService userService;
+    private final EventStartEventPublisher eventStartEventPublisher;
+
+    @Scheduled(fixedRate = 60000)
```
Думаю, что лучше это запускать на основании cron-выражения.
fixedRate запускается каждую минуту, начиная с момента запуска программы.
А cron-выражение будет описаться на системные часы ПК/Сервера.
Замечание не существенное, но всё же.

Soddik replied:
такие вещи лучше всего выносить проперти

DmitryRogulkin replied:
Перенес в крон app.yaml

antikler commented on src/main/java/school/faang/user_service/scheduler/EventStartScheduler.java:
```diff
@@ -0,0 +1,50 @@
+
+            EventStartEvent eventStartEvent = new EventStartEvent(
+                    String.valueOf(event.getId()),
+                    participants
```
Согласно задания:  событии должны содержаться id события, список id всех участников.
Не нужен список участников, нужны их id.

DmitryRogulkin replied:
Действительно. Исправляю.

Soddik commented on src/main/java/school/faang/user_service/config/context/RedisConfig.java:
```diff
@@ -0,0 +1,43 @@
+
+    @Bean
+    public JedisConnectionFactory jedisConnectionFactory() {
+        RedisStandaloneConfiguration config = new RedisStandaloneConfiguration(redisHost, redisPort);
```
конфигурация не полная, пул соединений

DmitryRogulkin replied:
переделал

Soddik commented on src/main/java/school/faang/user_service/config/context/RedisConfig.java:
```diff
@@ -0,0 +1,43 @@
+        RedisTemplate<String, Object> template = new RedisTemplate<>();
+        template.setConnectionFactory(jedisConnectionFactory);
+        template.setKeySerializer(new StringRedisSerializer());
+        template.setValueSerializer(new GenericJackson2JsonRedisSerializer()); // сериализация в JSON
```
коммент

DmitryRogulkin replied:
Убрал.

Soddik commented on src/main/java/school/faang/user_service/config/context/RedisConfig.java:
```diff
@@ -0,0 +1,43 @@
+    private String redisHost;
+
+    @Value("${spring.data.redis.port}")
+    private int redisPort;
```
снова примитив

DmitryRogulkin replied:
Переделал.

Soddik commented on src/main/java/school/faang/user_service/messaging/event/EventStartEvent.java:
```diff
@@ -0,0 +1,16 @@
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class EventStartEvent {
```
Serializable ?

DmitryRogulkin replied:
Теперь да )

Soddik commented on src/main/java/school/faang/user_service/messaging/publisher/EventStartEventPublisherImpl.java:
```diff
@@ -0,0 +1,20 @@
+
+    @Override
+    public void publish(EventStartEvent event) {
+        redisTemplate.convertAndSend(topic.getTopic(), event);
```
лучше обернуть в try catch

DmitryRogulkin replied:
добавил.

Soddik commented on src/main/java/school/faang/user_service/scheduler/EventStartScheduler.java:
```diff
@@ -0,0 +1,50 @@
+
+    @Scheduled(fixedRate = 60000)
+    public void publishStartEvents() {
+        List<Event> startingNow = eventRepository.findAll().stream()
```
не используй пост фильтр, фильтруй на уровне БД

DmitryRogulkin replied:
Хорошо, буду дергать данные в запросе SQL.

Soddik commented on src/main/java/school/faang/user_service/scheduler/EventStartScheduler.java:
```diff
@@ -0,0 +1,50 @@
+                    .map(user -> user.getId())
+                    .toList();
+
+            List<UserDto> participants = userService.getUsersByIds(participantIds);
```
если будет исключение, то обработка всех остальных событий прекратится

DmitryRogulkin replied:
Понял, переделаю.

Soddik commented on src/main/java/school/faang/user_service/scheduler/EventStartScheduler.java:
```diff
@@ -0,0 +1,50 @@
+                    participants
+            );
+
+            eventStartEventPublisher.publish(eventStartEvent);
```
если будет исключение, то обработка всех остальных событий прекратится

DmitryRogulkin replied:
Накину, логирование.

Soddik commented on src/main/java/school/faang/user_service/scheduler/EventStartScheduler.java:
```diff
@@ -0,0 +1,50 @@
+    private final EventStartEventPublisher eventStartEventPublisher;
+
+    @Scheduled(fixedRate = 60000)
+    public void publishStartEvents() {
```
подумай о разбиении на бачи или асинхронной обработке

DmitryRogulkin replied:
Подумаю, отпишу.

DmitryRogulkin committed:
https://faang-school.atlassian.net/browse/BJS2-65650

Update after comment TechLead
Edit RedisConfig and add RedisProperties

DmitryRogulkin committed:
https://faang-school.atlassian.net/browse/BJS2-65650

Update after comment TechLead
Edit EventStartEventPublisherImpl + EventStartEvent + EventStartScheduler + add SQL EventRepository

Soddik reviewed (dismissed):
Нужен фикс конфликтов

DmitryRogulkin committed:
Merge branch 'manticore-master-stream9' into feature-riko-BJS2-65650

DmitryRogulkin commented:
> Нужен фикс конфликтов

Поправил.

DmitryRogulkin committed:
https://faang-school.atlassian.net/browse/BJS2-65650
